### ISSUE TITLE
ClickHouse dialect: WHERE after ARRAY JOIN is parsed as alias, predicate becomes unparsable

### ISSUE THREAD

**Issue by Cayan** [sqlfluff/sqlfluff#7836]
**Search before asking:** I searched the issues and found no similar issues.

## What Happened

A `SELECT` that uses `ARRAY JOIN` followed by a `WHERE` clause fails to parse under the `clickhouse` dialect. The parser handles `ARRAY JOIN` itself, but does not terminate the clause. Instead it consumes the trailing `WHERE` keyword as an `alias_expression` on the `ARRAY JOIN`, leaving the predicate body as an unparsable section.

## Expected Behaviour

The query parses cleanly. ClickHouse explicitly permits `WHERE` (and `PREWHERE`) to follow `ARRAY JOIN`, and the [ARRAY JOIN docs](https://clickhouse.com/docs/sql-reference/statements/select/array-join) state that `ARRAY JOIN` must be specified before `WHERE/PREWHERE` in the `SELECT` structure.

## Observed Behaviour

```
$ sqlfluff lint --dialect clickhouse repro.sql
== [repro.sql] FAIL
L:   4 | P:   1 | RF04 | Keywords should not be used as identifiers. [references.keywords]
L:   4 | P:   7 |  PRS | Line 4, Position 7: Found unparsable section: "item != ''"
WARNING: Parsing errors found and dialect is set to 'clickhouse'. Have you configured your dialect correctly?
```

Both findings have the same root cause:
- **RF04 at `L:4 P:1`** is the symptom. The parser treats the `WHERE` keyword as a `naked_identifier` (an alias).
- **PRS at `L:4 P:7`** is the consequence. Once `WHERE` has been consumed as an alias, the predicate `item != ''` is left dangling with no clause to live under.

The parse tree (from `sqlfluff parse --dialect clickhouse repro.sql`) shows this directly:

```
            from_clause:
                ...
                from_expression:
                    from_expression_element:
                        table_expression:
                            table_reference:
                                naked_identifier:             'some_table'
                    array_join_clause:
                        keyword:                              'ARRAY'
                        keyword:                              'JOIN'
                        select_clause_element:
                            column_reference:
                                naked_identifier:             'line_items'
                            alias_expression:
                                alias_operator:
                                    keyword:                  'AS'
                                naked_identifier:             'item'
                    alias_expression:
                        naked_identifier:                     'WHERE'   <-- bug
            unparsable:
                word:                                         'item'
                not:                                          '!'
                equals:                                       '='
                single_quote:                                 "''"
```

The fix likely lives in the `ArrayJoinClauseSegment` grammar, or in how it integrates into `FromClauseSegment` / `SelectStatementSegment`. The clause needs a terminator that prevents alias matching on a following `WHERE` / `PREWHERE` / `GROUP BY` / `HAVING` / `QUALIFY` / `ORDER BY` / `LIMIT` / `SETTINGS` keyword.

## How to reproduce

Save as `repro.sql`:

```sql
SELECT item
FROM some_table
ARRAY JOIN line_items AS item
WHERE item != ''
```

Run:

```
sqlfluff lint --dialect clickhouse repro.sql
```

**Control case** (removing the `WHERE`) parses cleanly, confirming the boundary:

```sql
SELECT item
FROM some_table
ARRAY JOIN line_items AS item
```

```
$ sqlfluff lint --dialect clickhouse control.sql
All Finished!
```

## Dialect

`clickhouse`

## Version

```
sqlfluff, version 4.1.0
Python 3.13
```

## Configuration

None. Reproduces with stock `--dialect clickhouse` and no `.sqlfluff` file.

## Code of Conduct

I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md).

**Comment by Cayan** [comment]
Quick follow-up to narrow the fix location: the actual bug is in `JoinLikeClauseGrammar` at [`src/sqlfluff/dialects/dialect_clickhouse.py:208`](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_clickhouse.py#L208), one level up from `ArrayJoinClauseSegment` itself.

The grammar is:

```python
JoinLikeClauseGrammar=Sequence(
    AnyNumberOf(
        Ref("ArrayJoinClauseSegment"),
        min_times=1,
    ),
    Ref("AliasExpressionSegment", optional=True),   # <-- this swallows WHERE
),
```

The trailing `Ref("AliasExpressionSegment", optional=True)` matches the following `WHERE` keyword as a `naked_identifier` alias.

Per the [ClickHouse ARRAY JOIN syntax](https://clickhouse.com/docs/sql-reference/statements/select/array-join), aliases belong on individual array elements (already handled by the `Delimited(Ref("SelectClauseElementSegment"))` inside `ArrayJoinClauseSegment`), not on the clause itself. A spot check of `test/fixtures/dialects/clickhouse/join.sql` confirms zero existing fixtures rely on a clause-level alias.

Local one-line removal of that `Ref("AliasExpressionSegment", optional=True)` clears the repro and the three real-world dbt models that triggered the original report. Happy to open a PR if useful — running the full test suite + fixture regeneration first.
